### DISCUSSION TITLE
publicUrl is always undefined but data.publicUrl display the url

### DISCUSSION THREAD

**Discussion by unit-002** [supabase/supabase#14026] [Questions] [answered]
I am new to supabase and I am still learning to integrate supabase to nextjs 13. Just a question regarding retrieving url from storage bucket. In the documentation the sample code is:

```
const { publicURL, error } = supabase
  .storage
  .from('public-bucket')
  .getPublicUrl('folder/avatar1.png')
```
if I `console.log(publicURL)` it shows undefined but from older samples using:

```
const { data, error } = supabase
  .storage
  .from('public-bucket')
  .getPublicUrl('folder/avatar1.png')
```
if I `console.log(data.publicURL)` it displays the url

question is how come they are different? one works and the other doesn't?

**Comment by silentworks** [comment, marked as answer]
Hey @unit-002, the first example you've shown is from the now deprecated version 1 of the supabase-js library https://supabase.com/docs/reference/javascript/v1/storage-from-getpublicurl, while the second example is from version 2 of the supabase-js library https://supabase.com/docs/reference/javascript/storage-from-getpublicurl.

**Reply by hbneurons** [reply]
I see I was looking at the wrong page. I have to get used to the website. Thanks a lot.
